core/src/main/kotlin/org/evomaster/core/search/gene/NumberGene.kt

```diff
@@ -25,4 +30,15 @@ abstract class NumberGene<T : Number>(name: String, var value: T) : Gene(name, m
             value.toFloat()
 
 
+    /**
+     * @return whether the [value] is between [min] and [max] if they are specified
+     */
+    override fun isValid() : Boolean{
+        if (max != null && value.toDouble() > max!!.toDouble())
+            return false
```

Comment:
how is that possible? shouldn't this prevented in the sampler/mutator? 
or is this method just used for debugging / assertion writing?

Reply:
it is only used for testing purposes to check whether the value is [min, max] when they are specified.

Reply:
in that case, add such clarification in the comments of the `isValid` declaration in super-class